R/LSbinaryclassification.R

```diff
@@ -0,0 +1,1291 @@
+                          ) +
+    ggplot2::geom_abline(slope = 1, intercept = 0, linetype = 2)
+
+  if(options[["credibleInterval"]])
+    plot <- plot + ggplot2::geom_ribbon(data    = ciData,
+                                        mapping = ggplot2::aes(x=x,ymin=ymin,ymax=ymax),
+                                        alpha   = 0.5, inherit.aes = FALSE)
+
+  plot <- plot +
+    ggplot2::geom_line(size = 1.5) +
+    jaspGraphs::geom_point(data = pointData, size = 5) +
+    ggplot2::xlab(gettext("False positive rate (1-Specificity)")) +
+    ggplot2::ylab(gettext("True positive rate (Sensitivity)"))
+
+
+  plot <- jaspGraphs::themeJasp(plot)
+
+  return(plot)
+}
+
+.bcFillPlotROC.bcData <- function(results, dataset, options) {
+
+  thresholds <- c(dataset$marker, options[["threshold"]])
+
+  data <- data.frame(
+    fpr = numeric(length(thresholds)+2), tpr = numeric(length(thresholds)+2)
+  )
+  data[1,] <- c(0, 0)
+  data[2,] <- c(1, 1)
+  for(i in seq_along(thresholds)+2) {
+    res <- coef(.bcGetPosterior(options = options, dataset = dataset, threshold = thresholds[i]))
+    data[i,] <- unlist(res[c("falsePositiveRate", "sensitivity")])
+  }
+
+  summ <- summary(results, ciLevel = sqrt(options[["ciLevel"]]))
+  pointData <- data.frame(fpr = summ["falsePositiveRate", "estimate"],
+                          tpr = summ["sensitivity",           "estimate"])
+  ciData <- data.frame(x    = unlist(summ["falsePositiveRate", c("lowerCI", "upperCI")]),
+                       ymin = summ["sensitivity", "lowerCI"],
+                       ymax = summ["sensitivity", "upperCI"])
+
+  plot <- ggplot2::ggplot(data    = data,
+                          mapping = ggplot2::aes(x = fpr, y = tpr)
+                          ) +
+    ggplot2::geom_abline(slope = 1, intercept = 0, linetype = 2)
+
+  if(options[["credibleInterval"]])
+    plot <- plot + ggplot2::geom_ribbon(data    = ciData,
+                                        mapping = ggplot2::aes(x=x,ymin=ymin,ymax=ymax),
+                                        alpha   = 0.5, inherit.aes = FALSE)
+
+  plot <- plot +
+    ggplot2::geom_step(size = 1.5) +
+    jaspGraphs::geom_point(data = pointData, size = 5) +
+    ggplot2::xlab(gettext("False positive rate (1-Specificity)")) +
+    ggplot2::ylab(gettext("True positive rate (Sensitivity)"))
+
+
+  plot <- jaspGraphs::themeJasp(plot)
+
+  return(plot)
+}
+
+## Test characteristics ----
+.bcPlotTestCharacteristics <- function(results, plotsContainer, dataset, options, ready, position) {
+  if( isFALSE(options[["plotTestCharacteristics"]])     ) return()
+  if(!is.null(plotsContainer[["plotTestCharacteristics"]]) ) return()
+
+  plotsContainer[["plotTestCharacteristics"]] <-
+    createJaspPlot(title        = gettext("Test characteristics"),
+                   dependencies = c("plotTestCharacteristics", "credibleInterval", "ciLevel"),
+                   position     = position,
+                   width        = 500
```

Comment:
Maybe make all these plots the same size?
![image](https://user-images.githubusercontent.com/25059399/116854585-35560100-abf8-11eb-8619-61de6dc157dc.png)